Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp

```diff
@@ -368,6 +368,37 @@ static inline float normalizeGXWeight(float value)
     return 523.7 * value - 109.3;
 }
 
+// These values were experimentally gathered from the various named weights of San Francisco.
+static struct {
+    float ctWeight;
+    float cssWeight;
+} keyframes[] = {
+    { -0.8, 30 },
+    { -0.4, 274 },
+    { 0, 400 },
+    { 0.23, 510 },
+    { 0.3, 590 },
+    { 0.4, 700 },
+    { 0.56, 860 },
+    { 0.62, 1000 },
+};
+static_assert(WTF_ARRAY_LENGTH(keyframes) > 0);
+
+float normalizeCTWeight(float value)
+{
+    if (value < keyframes[0].ctWeight)
+        return keyframes[0].cssWeight;
+    for (size_t i = 0; i < WTF_ARRAY_LENGTH(keyframes) - 1; ++i) {
+        auto& before = keyframes[i];
+        auto& after = keyframes[i + 1];
+        if (value < before.ctWeight || value > after.ctWeight)
+            continue;
```

Comment:
This condition took me too long to work out. :-) It would be easier to understand as:

```c++
if (value >= before.ctWeight && value <= after.ctWeight) {
    ...
    return ...
}
```

Reply:
I think in general in WebKit we prefer early-return style, but I'm happy to reorganize this code to make it more readable.